GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift

```diff
@@ -56,26 +48,72 @@ public class GiniSwitchSdk {
         self.clientId = clientId
         self.clientSecret = clientSecret
         self.clientDomain = domain
-        GiniSwitchSdkStorage.activeSwitchSdk = self     // TODO: Don't use GiniSwitchSdkStorage - find a better solution
+        extractionsManager = ExtractionsManager(clientId: clientId, clientSecret: clientSecret, clientDomain: clientDomain)
+        extractionsManager.delegate = self
     }
     
     /*
      * Convenience method for creating the SDK's UI
      */
     public func instantiateSwitchViewController() -> UIViewController {
-        return userInterface.initialViewController
+        coordinator = MultiPageCoordinator(extractionsManager: extractionsManager, onboarding: configuration.onboarding)
+        coordinator?.delegate = self
+        return coordinator!.initialViewController
     }
     
     public func sendFeedback(_ feedback:ExtractionCollection) {
-        feedbackHandler(feedback)
+        extractionsManager.sendFeedback(feedback)
+    }
+
+}
+
+extension GiniSwitchSdk: ExtractionsManagerDelegate {
+    
+    func extractionsManager(_ manager:ExtractionsManager, didEncounterError error:NSError) {
+        // check if it's an error that needs to be shown to the user
+        if error.isHumanReadable {
+            coordinator?.displayError(error)
+        }
+        delegate?.switchSdk(self, didReceiveError: error)
     }
     
-    /*
-     * Disposes of the current SDK object. Should be called after clients are done with the SDK
-     * in order to clean up
-     */
-    public func terminate() {
-        GiniSwitchSdkStorage.activeSwitchSdk = nil     // TODO: Don't use GiniSwitchSdkStorage - find a better solution
+    func extractionsManager(_ manager:ExtractionsManager, didChangePageCollection collection:PageCollection) {
+        coordinator?.refreshPagesCollectionView()
+    }
+    
+    func extractionsManager(_ manager:ExtractionsManager, didChangeExtractions extractions:ExtractionCollection) {
+        self.extractions = extractions
+        delegate?.switchSdk(self, didChangeExtractions: extractions)
+    }
+    
+    func extractionsManagerDidAuthenticate(_ manager:ExtractionsManager) {
+        
+    }
+    
+    func extractionsManagerDidCreateOrder(_ manager:ExtractionsManager) {
+        
+    }
+    
+    func extractionsManagerDidCompleteExtractions(_ manager:ExtractionsManager) {
+        // Don't interrupt the user immediately. Allow them to complete the last action
+        // before notifying them that extractions are complete
```

Comment:
It's this thing that the "Extractions Complete" (Splash) screen is only shown after the review screen. The thing you don't have on Android. So if we see that all extractions are there, we just mark them so and wait for the user to take another pic.